sycl/doc/extensions/proposed/sycl_ext_intel_fpga_pipes_properties.asciidoc

```diff
@@ -0,0 +1,280 @@
+| Valid values: Non-negative integer value.
+
+Default value: 0
+
+User defined minimum number of words in units of data type size that the pipe
+must be able to store without any being read out. A minimum capacity is required
+in some algorithms to avoid deadlock, or for performance tuning. An
+implementation can include more capacity than this parameter, but not less.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+
+|`ready_latency`
+| Valid values: Non-negative integer value.
+
+Default value: 0
+
+The number of cycles between when the ready signal is deasserted and when the
+pipe can no longer accept new inputs.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`bits_per_symbol`
+| Valid values: A positive integer value that evenly divides by the data type size. 
+
+Default value: Datatype size
+
+Describes how the data is broken into symbols on the data bus.
+
+Data is broken down according to how you set the first_symbol_in_high_order_bits
+property. By default, data is broken down in little endian order.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`uses_valid`
+| Valid values: true or false
+
+Default value: true
+
+Controls whether a valid signal is present on the pipe interface. If false, the
+upstream source must provide valid data on every cycle that ready is asserted.
+
+This is equivalent to changing the pipe read calls to tryRead and assuming that
+success is always true.
+
+If set to false, min_capacity and ready_latency must be 0.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`uses_ready`
+| Valid values: true or false
+
+Default value: true
+
+Controls whether a ready signal is present. If false, the downstream sink must
+be able to accept data on every cycle that valid is asserted. This is
+equivalent to changing the pipe read calls to tryWrite and assuming that success
+is always true.
+
+If set to false, ready_latency must be 0.
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`in_csr`
+| Valid Values: true or false
+
+Default Value: false
+
+Controls whether the host pipe is implemented using the Control and Status register (CSR).
+
+This property is not guaranteed to be respected if the pipe is an inter-kernel
+pipe. The compiler is allowed to optimize the pipe if both sides are visible.
+
+|`first_symbol_in_high_order_bits`
+| Valid values: true or false
+
+Default value: false
+
+Specifies whether the data symbols in the pipe are in big-endian
```

Comment:
By saying "the data symbols in the pipe" it sounds like we're referring to how symbols are order across the depth of the pipe, which is certainly not our intention.  We should make it clear that this only refers to the ordering of symbols within a given word/transaction.  Some alternatives are:

... the data symbols in each word written to/read from the pipe are ...
... the data symbols in each transaction are ...
... the data symbols in the data channel of the pipe are ...
The first_symbol_in_high_order_bits property of the stream as defined in the Avalon-ST specification.